vault/programs/vault/src/context.rs

```diff
@@ -0,0 +1,419 @@
+    )]
+    pub vault: Box<Account<'info, Vault>>,
+    #[account(
+        seeds = [VAULT_AUTHORITY_SEED.as_bytes(), vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault_authority
+    )]
+    pub vault_authority: AccountInfo<'info>,
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    #[account(
+        mut,
+        seeds = [REDEEMABLE_MINT_SEED.as_bytes(), vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.redeemable_mint
+    )]
+    pub redeemable_mint: Box<Account<'info, Mint>>,
+    #[account(
+        mut,
+        seeds = [VAULT_USDC_SEED.as_bytes(), vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault_usdc
+    )]
+    pub vault_usdc: Box<Account<'info, TokenAccount>>,
+    // Programs and Sysvars
+    pub token_program: Program<'info, Token>,
+}
+
+#[derive(Accounts)]
+pub struct ExchangeRedeemableForUsdc<'info> {
+    // User Accounts
+    #[account(mut)]
+    pub user_authority: Signer<'info>,
+    #[account(
+        mut,
+        constraint = user_usdc.owner == user_authority.key(),
+        constraint = user_usdc.mint == usdc_mint.key()
+    )]
+    pub user_usdc: Box<Account<'info, TokenAccount>>,
+    #[account(
+        mut,
+        seeds = [USER_REDEEMABLE_SEED.as_bytes(),
+            vault.vault_name.as_ref().strip(),
+            user_authority.key().as_ref()],
+        bump
+    )]
+    pub user_redeemable: Box<Account<'info, TokenAccount>>,
+    // vault Accounts
+    #[account(
+        seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault,
+        has_one = usdc_mint
+    )]
+    pub vault: Box<Account<'info, Vault>>,
+    #[account(
+        seeds = [VAULT_AUTHORITY_SEED.as_bytes(), vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault_authority
+    )]
+    pub vault_authority: AccountInfo<'info>,
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    #[account(
+        mut,
+        seeds = [REDEEMABLE_MINT_SEED.as_bytes(), vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.redeemable_mint
+    )]
+    pub redeemable_mint: Box<Account<'info, Mint>>,
+    #[account(
+        mut,
+        seeds = [VAULT_USDC_SEED.as_bytes(), vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault_usdc
+    )]
+    pub vault_usdc: Box<Account<'info, TokenAccount>>,
+    // Programs and Sysvars
+    pub token_program: Program<'info, Token>,
+}
+
+#[derive(Accounts)]
+pub struct WithdrawVaultUsdc<'info> {
+    // vault Authority Accounts
+    pub vault_admin: Signer<'info>,
+    // Doesn't need to be an ATA because it might be a DAO account
+    #[account(
+        mut,
+        constraint = vault_admin_usdc.owner == vault_admin.key(),
+        constraint = vault_admin_usdc.mint == usdc_mint.key()
+    )]
+    pub vault_admin_usdc: Box<Account<'info, TokenAccount>>,
+    // vault Accounts
+    #[account(
+        seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault,
+        has_one = vault_admin,
```

Comment:
Like this - I'm not sure what this check is trying to assert  - would much rather 

constraint = vault.... = vault_admin

Reply:
Ok fine I'll cave in